controllers/imagerepository_notifications.go

```diff
@@ -0,0 +1,173 @@
+		r.QuayOrganization,
+		imageRepository.Spec.Image.Name,
+		quay.Notification{
+			Title:  notification.Title,
+			Event:  string(notification.Event),
+			Method: string(notification.Method),
+			Config: quay.NotificationConfig{
+				Url: notification.Config.Url,
+			},
+			EventConfig: quay.NotificationEventConfig{},
+		})
+
+	if err != nil {
+		return imagerepositoryv1alpha1.NotificationStatus{}, err
+	}
+	return imagerepositoryv1alpha1.NotificationStatus{
+		UUID:  quayNotification.UUID,
+		Title: notification.Title,
+	}, err
+}
+
+func (r *ImageRepositoryReconciler) HandleNotifications(ctx context.Context, imageRepository *imagerepositoryv1alpha1.ImageRepository) error {
+	log := ctrllog.FromContext(ctx).WithName("HandleNotifications")
+
+	if imageRepository.Status.Notifications == nil {
+		// No status notifications to check
+		return nil
+	}
+	allNotifications, err := r.QuayClient.GetNotifications(r.QuayOrganization, imageRepository.Spec.Image.Name)
+	if err != nil {
+		if err = r.UpdateImageRepositoryStatusMessage(ctx, imageRepository, "Couldn't retrieve all Quay notifications"); err != nil {
+			return err
+		}
+		return nil
+	}
+	for index := 0; index < len(imageRepository.Status.Notifications); index++ {
+		statusNotification := &imageRepository.Status.Notifications[index] // This way we can update the UUID if notification gets updated
+		existsInSpec := false
+		for _, notification := range imageRepository.Spec.Notifications {
+			if notification.Title == statusNotification.Title {
+				existsInSpec = true
+				quayNotification := notificationExistsInQuayByUUID(statusNotification.UUID, imageRepository, allNotifications)
+				if isNotificationChanged(notification, quayNotification) {
+					// Updated item in Spec.Notifications: update notification in Quay
+					log.Info("Updating notification in Quay", "Title", notification.Title, "Event", notification.Event, "Method", notification.Method, "UUID", quayNotification.UUID)
+					updatedNotification, err := r.QuayClient.UpdateNotification(
+						r.QuayOrganization,
+						imageRepository.Spec.Image.Name,
+						statusNotification.UUID,
+						quay.Notification{
+							Title:  notification.Title,
+							Event:  string(notification.Event),
+							Method: string(notification.Method),
+							Config: quay.NotificationConfig{
+								Url: notification.Config.Url,
+							},
+							EventConfig: quay.NotificationEventConfig{},
+						})
+					if err != nil {
+						log.Error(err, "failed to update notification", "Title", statusNotification.Title, "UUID", statusNotification.UUID)
+						if err = r.UpdateImageRepositoryStatusMessage(ctx, imageRepository, "Error while updating the notification in Quay"); err != nil {
+							return err
+						}
+						return nil
+					}
+					statusNotification.UUID = updatedNotification.UUID
+				}
+			} else {
+				quayNotification := notificationExistsInQuayByTitle(notification.Title, imageRepository, allNotifications)
+				if quayNotification.UUID == "" {
```

Comment:
If the notification does not exists, it's better to return `nil` from `notificationExistsInQuayByTitle`. Then `quayNotification.UUID == ""` will not be ambiguous.
Moreover, if found notification data is not used, it's even better to return `bool`